packages/fluentui/react-builder/src/config.tsx

```diff
@@ -0,0 +1,717 @@
+    if (tree.props.children.find(jte => typeof jte !== 'string' && jte.uuid === uuidFromKey)) {
+      return tree;
+    }
+    for (let i = 0; i < tree?.props?.children.length && ret === null; ++i) {
+      const e = tree?.props?.children[i];
+      if (typeof e !== 'string') {
+        ret = jsonTreeFindParent(e, uuidFromKey);
+      }
+    }
+  }
+  return ret;
+};
+
+export const jsonTreeDeleteElement = (tree: JSONTreeElement, uuid: string | number): JSONTreeElement => {
+  const uuidFromKey = keyToUUID(uuid);
+
+  const omitChildWithUuid = (tree: JSONTreeElement, uuid: string | number): JSONTreeElement => {
+    if (Array.isArray(tree?.props?.children)) {
+      tree.props.children = tree.props.children
+        .filter(jte => typeof jte === 'string' || jte.uuid !== uuid)
+        .map(jte => (typeof jte === 'string' ? jte : omitChildWithUuid(jte, uuid)));
+    }
+
+    return tree;
+  };
+
+  if (tree.uuid === uuidFromKey) {
+    return null;
+  }
+
+  return omitChildWithUuid(tree, uuid);
+};
+
+export const jsonTreeCloneElement = (tree: JSONTreeElement, element: any): JSONTreeElement => {
+  const result = _.transform(element, (acc, value, key) => {
+    if (_.isPlainObject(value)) {
+      acc[key] = jsonTreeCloneElement(tree, value);
+    } else if (Array.isArray(value)) {
+      acc[key] = jsonTreeCloneElement(tree, value);
+    } else {
+      acc[key] = value;
+    }
+  }) as any;
+
+  if (_.isPlainObject(result) && result.$$typeof === 'Symbol(react.element)' && result.uuid) {
+    result.uuid = getUUID();
+  }
+  return result;
+};
+
+/**
+ * Displays a knob with the ability to switch between data `types`.
+ */
+export const MultiTypeKnob: React.FunctionComponent<{
+  label: string;
+  types: ('boolean' | 'number' | 'string' | 'literal')[];
+  value: any;
+  onChange: (value: any) => void;
+  literalOptions: string[];
+}> = ({ label, types, value, onChange, literalOptions }) => {
+  const valueType = typeof value;
+  const defaultType = valueType !== 'undefined' ? valueType : types[0];
+  const [type, setType] = React.useState(defaultType);
+  const knob = knobs[type];
+  const handleChangeType = React.useCallback(
+    e => setType(e.target.value), // @ts-ignore
+    [],
+  );
+
+  // console.log('MultiTypeKnob', { label, value, type, types });
+
+  return (
+    <div style={{ paddingBottom: '4px', marginBottom: '4px', opacity: knob ? 1 : 0.4 }}>
+      <div>
+        {type !== 'boolean' && <label>{label} </label>}
+        {types.length === 1 ? (
+          <code style={{ float: 'right' }}>{type}</code>
+        ) : (
+          types.map(t => (
+            <button key={t} onClick={() => handleChangeType(t)}>
+              {t}
+            </button>
+          ))
+        )}
+      </div>
+      {knob && knob({ options: literalOptions, value, onChange })}
+      {type === 'boolean' && <label> {label}</label>}
+    </div>
+  );
+};
+
+export const knobs = {
+  boolean: ({ value, onChange }) => (
```

Comment:
pass a unique id to each knob, use it in `<input id={...} />` so that it can be paired with the label